### PR TITLE
Update incident

### LINKED ISSUE #11: Update existing Incident (PUT Incident)

### PR DESCRIPTION
**Pull request by amsmyth1**
### What Changed?
- Create end point for Updating Incidents
### What's Next?
- Refactor to Facade Design Pattern
- Add edge case testing for incorrect routes (like PATCH 'api/v1/incidents/')
### Known Issues?
- The validations of the params need to be refactored
### Test Coverage?
- 100%, happy, sad and edge case tested
### README & Documentation
- Updated Readme to document API Contract for updating an incident
### Closed Issues
Close #11

### CODE REVIEW

**Review by jmichlitsch — APPROVED** [review]
Like what you're doing, I do agree we could refactor the valid params method but otherwise, great job.

File: spec/requests/api/v1/update_an_incident_spec.rb [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    end
+  end
+  describe "edge case" do
+    it "can't update an incident that does not exist" do
+      incident = create(:incident, active: false, name: "Jim Creeks Fire", incident_type: "Fire", description: "5th alarm fire", location: "Denver, CO", start_date: "2020-05-01", close_date: "2020-05-03")
+      incident_params = {close_date: ""}
+
+      headers = {"CONTENT_TYPE" => "application/json"}
+
+      patch "/api/v1/incidents/&&&", headers: headers, params: JSON.generate(incident_params)
```
Comment (klatour324) [inline spec/requests/api/v1/update_an_incident_spec.rb:111]: I love it! thank you for taking the suggestion into consideration here for edge case tesing :)

File: app/controllers/api/v1/incidents_controller.rb [hunk 2]
```diff
@@ -14,6 +14,21 @@ def create
     end
   end
 
+  def update
+    incident = Incident.where(id: params[:id])
+    if incident.count == 0
+      error = "Incident does not exist with that id"
+      render json: { error: error }, status: :not_found
+    else
+      if validate_update_params?
```
Comment (klatour324) [inline app/controllers/api/v1/incidents_controller.rb:23]: Would this just be an `elsif` statement like `elsif validate_update_params?`?

**Review by klatour324 — COMMENTED** [review]
Hey Alexa! Lots of detail here and great work. Thank you for taking my comment from the previous PR into consideration for edge case testing :). I just have one other comment here in regards to the update method conditional branching.

**Review by klatour324 — APPROVED** [review]
Woo! Glad it worked 😄
